backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/VmHandler.java

```diff
@@ -1434,12 +1434,7 @@ public void autoSelectDefaultDisplayType(Guid srcEntityId,
         }
 
         if (defaultDisplayType == null) {
-            if (!displayGraphicsSupport.isEmpty()) {// when not found otherwise, let's take osinfo's record as the default
-                Map.Entry<DisplayType, Set<GraphicsType>> entry = displayGraphicsSupport.entrySet().iterator().next();
-                defaultDisplayType = entry.getKey();
-            } else {// no osinfo record
-                defaultDisplayType = DisplayType.qxl;
-            }
+            defaultDisplayType = DisplayType.vga;
```

Comment:
using vga instead of qxl when there's no record in osinfo is ok but when there is we won't respect it this way..
imagine one modifies "devices.display.protocols" to vnc/bochs - we'd still use vga nevertheless which is not great
how about keeping the first block and pick something that is not qxl, if exists, in that case?

Reply:
i think this is what the current PR has. it will iterate over the given values from osinfo, if there is vnc/vga it will be preferred. however, it needs to be improved. 
based on michal comment below it suggest just to switch osinfo, having the original logic. WDYT? in my opinion it's a good option, and a pretty simple one.